File: src/main/scala/com/celestia/models/GameObject.scala

package com.celestia.models

import com.celestia.interfaces.{IRenderStrategy, IUpdateStrategy, IGameObject}

/**
  * Game Object class for holding state of an
  * individual game object. 
  */
abstract case class GameObject(
  override val position:Matrix3D,
  override val rotation:Matrix3D,
  override val scale:Matrix3D,

  override val updateStrategy:IUpdateStrategy,
  override val renderStrategy:IRenderStrategy 
) extends IGameObject

File: src/main/java/com/celestia/models/GameObject.java

package com.celestia.models;

public class GameObject {
    public final Matrix3D position; 
    public final Matrix3D rotation; 
    public final Matrix3D scale; 
    
    public final IRenderStrategy renderStrategy;
    public final IUpdateStrategy updateStrategy; 
    
    public GameObject(Matrix3D position, Matrix3D rotation, Matrix3D scale, IRenderStrategy renderStrategy, IUpdateStrategy updateStrategy) {
        this.position = position;
        this.rotation = rotation;
        this.scale = scale;
        this.renderStrategy = renderStrategy;
        this.updateStrategy = updateStrategy;
    }
}

